src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIteratorExtensions.cs

```diff
@@ -158,6 +164,37 @@ public static bool GetKnownMethod(this MemoryPoolIterator begin, out string know
         }
 
         /// <summary>
+        /// Checks 8 bytes from <paramref name="begin"/> that correspond to 'http://' or 'https://'
+        /// </summary>
+        /// <param name="begin">The iterator</param>
+        /// <param name="knownScheme">A reference to the known scheme, if the input matches any</param>
+        /// <returns>True when memory starts with known http or https schema</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool GetKnownHttpSchema(this MemoryPoolIterator begin, out string knownScheme)
+        {
+            knownScheme = null;
+            ulong value;
+            if (!begin.TryPeekLong(out value))
+            {
+                return false;
+            }
+
+            if ((value & _mask7Chars) == _httpSchemeLong)
```

Comment:
To make a distinction here, we support accepting absolute-form b/c the spec requires it and some **http clients** may send it. I'm in favor of rejecting anything with 'ws://' or 'wss:// because that is a buggy **WebSockets client**.